apps/remote_control/test/lexical/remote_control/code_intelligence/definition_test.exs

```diff
@@ -0,0 +1,370 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.DefinitionTest do
```

Comment:
This approach seems off to me. You have a project with files in it (I assume they're for the definitions), but you're also polluting the current VM with compiled artifacts via `Code.compile_string`. What are you aiming for?

Keeping this all encapsulated inside a project VM would make this somewhat easier, though I must admit, I'm not a huge fan of the fixtures being so far away from the testing code (but at the same time, there's stuff you just cannot reproduce in the current VM)

I suspect things are this way to make the tests run faster, but frankly, having a fast test that doesn't work like the prod code is of little value when you get to those edge cases.

Reply:
The purpose here is to make the test run faster, but you are right that fast tests that are incorrect have no value in edge cases. I have changed it to run on an isolated node.